src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,434 @@
+import { icon } from 'src/components/icons';
+import {
+  CriteriaType,
+  DatasetResponse,
+  DomainType,
+  ProgramDataListType,
+  ProgramDataRangeType,
+} from 'src/libs/ajax/DatasetBuilder';
+import colors from 'src/libs/colors';
+import * as Utils from 'src/libs/utils';
+import { PAGE_PADDING_HEIGHT, PAGE_PADDING_WIDTH } from 'src/pages/library/datasetBuilder/constants';
+import {
+  AnyCriteria,
+  Cohort,
+  CriteriaGroup,
+  DatasetBuilderState,
+  DomainCriteria,
+  newCriteriaGroup,
+  ProgramDataListCriteria,
+  ProgramDataRangeCriteria,
+} from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import { HomepageState, OnStateChangeType } from 'src/pages/library/datasetBuilder/DatasetBuilder';
+import { datasetBuilderCohorts } from 'src/pages/library/datasetBuilder/state';
+
+type CriteriaViewProps = { criteria: AnyCriteria; deleteCriteria: (criteria: AnyCriteria) => void };
+
+const CriteriaView = ({ criteria, deleteCriteria }: CriteriaViewProps) => {
+  return div(
+    {
+      style: {
+        display: 'flex',
+        width: '100%',
+        justifyContent: 'space-between',
+        alignItems: 'baseline',
+        paddingBottom: 5,
+        marginTop: 10,
+        marginBottom: 5,
+        borderBottom: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { margin: '5px', display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            'aria-label': 'delete criteria',
+            onClick: () => {
+              deleteCriteria(criteria);
+            },
+          },
+          [icon('minus-circle', { size: 24, style: { color: colors.danger() } })]
+        ),
+        div({ style: { marginLeft: 5 } }, [
+          Utils.cond(
+            [
+              'domainType' in criteria,
+              () => {
+                const domainCriteria = criteria as DomainCriteria;
+                return h(Fragment, [strong([`${domainCriteria.domainType.category}:`]), ` ${domainCriteria.name}`]);
+              },
+            ],
+            [
+              'listType' in criteria,
+              () => {
+                const listCriteria = criteria as ProgramDataListCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${listCriteria.value.name}`]);
+              },
+            ],
+            [
+              'rangeType' in criteria,
+              () => {
+                const rangeCriteria = criteria as ProgramDataRangeCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${rangeCriteria.low} - ${rangeCriteria.high}`]);
+              },
+            ],
+            [Utils.DEFAULT, () => div(['Unknown criteria type'])]
+          ),
+        ]),
+      ]),
+      `Count: ${criteria.count}`,
+    ]
+  );
+};
+
+export const renderCriteriaView = (deleteCriteria: (criteria: AnyCriteria) => void) => (criteria: AnyCriteria) =>
```

Comment:
I want to kill the pattern of renderXXXxthing with the sledgehammer of equivalent mass of the super-massive black hole at the center of our milky-way galaxy....

Please rework this and any similar repetition of this pattern to be a straight-issue React functional component, named CriteriaView and typed as React.FC<CriteriaViewProps> or something similar, and a CriteriaViewProps type defined just before it.